File: core/amber/src/main/python/core/architecture/handlers/modify_operator_logic_handler.py
from proto.edu.uci.ics.amber.engine.architecture.worker import ModifyOperatorLogicV2
from .handler_base import Handler
from ..managers.context import Context


class ModifyOperatorLogicHandler(Handler):
    cmd = ModifyOperatorLogicV2

    def __call__(self, context: Context, command: cmd, *args, **kwargs):
        context.operator_manager.update_operator(command.code, command.is_source)
        return None

File: core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/pythonworker/promisehandlers/ModifyOperatorLogicHandler.scala
package edu.uci.ics.amber.engine.architecture.pythonworker.promisehandlers

import edu.uci.ics.amber.engine.common.rpc.AsyncRPCServer.ControlCommand

object ModifyOperatorLogicHandler {
  final case class ModifyOperatorLogic(code: String, isSource: Boolean) extends ControlCommand[Unit]
}
